modules/Attendance/Services/MockAttendanceService.php

```
@@ -36,10 +37,69 @@ public function handleClockInProcess(ClockInDTO $clockInDTO, array $rawRequestDa
     {
         $user = auth()->user();
 
+        $timezone = getTimeZoneByRequest() ?? config('app.timezone');
+        $clockInCarbon = Carbon::parse($clockInDTO->getClockInTime(), $timezone);
+
+        // Get constraints for the clock-in date in the same timezone
+        $userConstraints = $this->userAttendanceService->getUserConstraints((string) $user->id, $clockInCarbon->format('Y-m-d'));
+
+        $canClockIn = false;
+        $activePeriod = null;
+
+        if (isset($userConstraints['work_rules']['all_work_periods'])) {
+            foreach ($userConstraints['work_rules']['all_work_periods'] as $period) {
+                $hasActiveAttendance = false;
+                if (!empty($period['attendance']) && is_array($period['attendance'])) {
+                    foreach ($period['attendance'] as $att) {
+                        if (($att['status'] ?? null) === 'active') {
+                            $hasActiveAttendance = true;
+                            break;
+                        }
+                    }
+                }
+
+                $periodDate = $period['date'] ?? $clockInCarbon->format('Y-m-d');
```

The `$periodDate` fallback to `$clockInCarbon->format('Y-m-d')` might mask potential data issues. Ensure this is the intended behavior.